packages/react-native-sdk/src/components/Livestream/ViewerLivestream/ViewerLobby.tsx

```diff
@@ -0,0 +1,168 @@
+import {
+  useCall,
+  useCallStateHooks,
+  useI18n,
+} from '@stream-io/video-react-bindings';
+import React, { useEffect, useMemo, useState } from 'react';
+import { ActivityIndicator, StyleSheet, Text, View } from 'react-native';
+import { useTheme } from '../../../contexts';
+import { CallingState } from '@stream-io/video-client';
+
+type LobbyProps = {
+  isLive: boolean;
+  handleJoinCall?: () => void;
+  setCallJoined: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+export const ViewerLobby = ({
+  isLive,
+  handleJoinCall,
+  setCallJoined,
+}: LobbyProps) => {
+  const styles = useStyles();
+  const { theme } = useTheme();
+  const { t } = useI18n();
+  const { useCallStartsAt, useParticipants } = useCallStateHooks();
+  const call = useCall();
+  const startsAt = useCallStartsAt();
+  const [error, setError] = useState<Error | undefined>(undefined);
+  const [countdown, setCountdown] = React.useState('');
+  const participants = useParticipants();
+
+  useEffect(() => {
+    setCallJoined(false);
+  }, [setCallJoined]);
```

Comment:
Most likely unnecessary `useEffect`. This data is already part of `call.state.callingState`.
Also, if `props.setCallJoined` function is declared inline, this effect will cause top level state to always be reset to `false`.